neqo-crypto/src/aead_null.rs

```diff
@@ -42,36 +42,60 @@ impl AeadNull {
     ) -> Res<&'a [u8]> {
         let l = input.len();
         output[..l].copy_from_slice(input);
-        output[l..l + 16].copy_from_slice(AEAD_NULL_TAG);
-        Ok(&output[..l + 16])
+        output[l..l + self.expansion()].copy_from_slice(AEAD_NULL_TAG);
+        Ok(&output[..l + self.expansion()])
     }
 
     #[allow(clippy::missing_errors_doc)]
-    pub fn decrypt<'a>(
-        &self,
-        _count: u64,
-        _aad: &[u8],
-        input: &[u8],
-        output: &'a mut [u8],
-    ) -> Res<&'a [u8]> {
-        if input.len() < AEAD_NULL_TAG.len() {
+    pub fn encrypt_in_place(&self, _count: u64, _aad: &[u8], data: &mut [u8]) -> Res<usize> {
+        let pos = data.len() - self.expansion();
+        data[pos..].copy_from_slice(AEAD_NULL_TAG);
+        Ok(data.len())
+    }
+
+    fn decrypt_check(&self, _count: u64, _aad: &[u8], input: &[u8]) -> Res<usize> {
+        if input.len() < self.expansion() {
             return Err(Error::from(SEC_ERROR_BAD_DATA));
         }
 
-        let len_encrypted = input.len() - AEAD_NULL_TAG.len();
+        let len_encrypted = input.len() - self.expansion();
```

Comment:
If you change this to a checked subtraction, you can use 

```suggestion
        let len_encrypted = input.len().checked_sub(self.expansion())
          .ok_or(Error::from(SEC_ERROR_BAD_DATA))?;
```